src/test/java/net/snowflake/ingest/streaming/internal/ParquetBufferTest.java

```diff
@@ -0,0 +1,278 @@
+    // scale in RowBufferTest.generateCol is set to 0, hence MILLIS
+    Assert.assertEquals(
+        LogicalTypeAnnotation.timestampType(false, LogicalTypeAnnotation.TimeUnit.MILLIS),
+        result.getLogicalTypeAnnotation());
+    Assert.assertEquals(Type.Repetition.OPTIONAL, result.getRepetition());
+  }
+
+  @Test
+  public void buildFieldTimestampTzSB16() {
+    ColumnMetadata testCol = RowBufferTest.generateCol("SB16", "TIMESTAMP_TZ", true);
+    Type result = this.rowBufferOnErrorContinue.getColumnParquetType(testCol, 0);
+
+    Assert.assertEquals("testCol", result.getName());
+    Assert.assertEquals(16, result.asPrimitiveType().getTypeLength());
+    Assert.assertEquals(0, result.asPrimitiveType().getId().intValue());
+
+    Assert.assertEquals(
+        PrimitiveType.PrimitiveTypeName.FIXED_LEN_BYTE_ARRAY,
+        result.asPrimitiveType().getPrimitiveTypeName());
+    Assert.assertEquals(
+        LogicalTypeAnnotation.decimalType(testCol.getScale(), 38),
+        result.getLogicalTypeAnnotation());
+    Assert.assertEquals(Type.Repetition.OPTIONAL, result.getRepetition());
+  }
+
+  @Test
+  public void buildFieldDate() {
+    ColumnMetadata testCol = RowBufferTest.generateCol("SB8", "DATE", true);
+    Type result = this.rowBufferOnErrorContinue.getColumnParquetType(testCol, 0);
+
+    Assert.assertEquals("testCol", result.getName());
+    // Assert.assertEquals(4, result.asPrimitiveType().getTypeLength());
+    Assert.assertEquals(0, result.asPrimitiveType().getId().intValue());
+
+    Assert.assertEquals(
+        PrimitiveType.PrimitiveTypeName.INT32, result.asPrimitiveType().getPrimitiveTypeName());
+    Assert.assertEquals(LogicalTypeAnnotation.dateType(), result.getLogicalTypeAnnotation());
+    Assert.assertEquals(Type.Repetition.OPTIONAL, result.getRepetition());
+  }
+
+  @Test
+  public void buildFieldTimeSB4() {
+    ColumnMetadata testCol = RowBufferTest.generateCol("SB4", "TIME", true);
+    Type result = this.rowBufferOnErrorContinue.getColumnParquetType(testCol, 0);
+
+    Assert.assertEquals("testCol", result.getName());
+    // Assert.assertEquals(4, result.asPrimitiveType().getTypeLength());
+    Assert.assertEquals(0, result.asPrimitiveType().getId().intValue());
+
+    Assert.assertEquals(
+        PrimitiveType.PrimitiveTypeName.INT32, result.asPrimitiveType().getPrimitiveTypeName());
+    // scale in RowBufferTest.generateCol is set to 0, hence MILLIS
+    Assert.assertEquals(
+        LogicalTypeAnnotation.timeType(false, LogicalTypeAnnotation.TimeUnit.MILLIS),
+        result.getLogicalTypeAnnotation());
+    Assert.assertEquals(Type.Repetition.OPTIONAL, result.getRepetition());
+  }
+
+  @Ignore
+  @Test
+  public void buildFieldTimeSB8() {
+
+    ColumnMetadata testCol = RowBufferTest.generateCol("SB8", "TIME", true);
+    Type result = this.rowBufferOnErrorContinue.getColumnParquetType(testCol, 0);
+    // -> exception java.lang.IllegalStateException: TIME(MILLIS,false) can only annotate INT32
```

Comment:
We should check if this time annotation is necessary for correct scanning.
If it is not then we should not generate it in this case.
I think it is related to some Parquet limitations for data types, e.g. for Iceberg,
but that is not relevant to pure Snowflake tables.
I guess `MILLIS` (SF scale <=3) have to be always sb4 in Parquet but our scanner does not need this annotation. The unit is derived from scale, even if it is int64.

Reply:
Could you check this?

Reply:
Where in XP should I look for this information?

Reply:
Is this what you meant https://github.com/snowflakedb/snowflake-ingest-java/blob/d3e91a01fe8e76a95269cd29bb73c3e51b73dab2/src/main/java/net/snowflake/ingest/streaming/internal/ParquetTypeGenerator.java#L239?

Reply:
Yes, looks so, you can check in debugger where it fails.

Reply:
The snippet above includes my change.
It was failing in `LogicalTypeAnnotation.timeType(false, timeUnit)`.
So I added ` (physicalType == AbstractRowBuffer.ColumnPhysicalType.SB8
                  ? LogicalTypeAnnotation.decimalType(scale, 18)`.

Reply:
ah I see
I would just use always `LogicalTypeAnnotation.decimalType` for now and remove all `LogicalTypeAnnotation.time*` because they look faulty and not needed for scanning. The other time units could be applied for sb8 but this logic will complicate things for now.

Reply:
ok. Should I use always `LogicalTypeAnnotation.decimalType(scale, 38)` 

 ```
      if (physicalType == AbstractRowBuffer.ColumnPhysicalType.SB16) {
      LogicalTypeAnnotation typeAnnotation = LogicalTypeAnnotation.decimalType(scale, 38);
      return Types.primitive(type, repetition).as(typeAnnotation).length(16).id(id).named(name);
    } else {
      LogicalTypeAnnotation typeAnnotation =
         LogicalTypeAnnotation.decimalType(scale, 18);
      return Types.primitive(type, repetition).as(typeAnnotation).id(id).named(name);
  }
```
or does the precision depend on the ColumnPhysicalType.SB8/4/2/1?

Reply:
Yes, I think so
it is the total number of decimal digits allowed:
https://docs.snowflake.com/en/sql-reference/data-types-numeric.html#number
every SBx type has a limit that it can hold